app/lib/server/functions/notifications/mobile.js

```diff
@@ -63,14 +72,52 @@ export async function sendSinglePush({ room, message, userId, receiverUsername,
 			messageType: message.t,
 			messageId: message._id,
 		},
-		roomName: settings.get('Push_show_username_room') && roomTypes.getConfig(room.t).isGroupChat(room) ? `#${ roomTypes.getRoomName(room.t, room) }` : '',
+		roomName:
+			settings.get('Push_show_username_room') && room.t !== 'd'
+				? `#${ roomTypes.getRoomName(room.t, room) }`
+				: '',
 		username,
 		message: settings.get('Push_show_message') ? notificationMessage : ' ',
 		badge: await getBadgeCount(userId),
 		usersTo: {
 			userId,
 		},
-		category: enableNotificationReplyButton(room, receiverUsername) ? CATEGORY_MESSAGE : CATEGORY_MESSAGE_NOREPLY,
+		category: enableNotificationReplyButton(room, receiverUsername)
+			? CATEGORY_MESSAGE
+			: CATEGORY_MESSAGE_NOREPLY,
+	});
+
+	const gcmKey = settings.get('Push_gcm_api_key');
+	const vapidPublic = settings.get('Vapid_public_key');
+	const vapidPrivate = settings.get('Vapid_private_key');
+
+	webpush.setGCMAPIKey(gcmKey);
+	webpush.setVapidDetails(
+		'https://www.viasat.com',
```

Comment:
This should not be hardcoded

Reply:
What key name should I give to this value? It represents the owner site url. Any suggestions?

Reply:
As per function documentation, it's subject. We can use the same. `vapidSubject` and `vapid_subject`
https://github.com/web-push-libs/web-push/blob/fbaeb12916233edb7acd6706575eb8e1e3a542ce/src/web-push-lib.js#L53

What is this `aes128gcm` fourth argument for? In the library I can see only three arguments are expected.

Reply:
aes128gcm is specifying the encryption type of payload. This is supported by all browsers so added this over here.